##################################################################################
$Id: xfree-4.2.x.-bison.fixes.patch,v 1.1 2002/12/05 22:35:11 azarah Exp $
##################################################################################

			     XFREE86 PATCH REPORT
			     ====================

Xfree86-Release: 4.2.x
Patch-ID: xc-giffordj-001

Bug-Reported-by:    giffordj@linkline.com
Assisted by    :    akim@epita.fr (Bison Support)
Patch by       :    giffordj@linkline.com

Bug-Description: Bison errors in xftgram.y gram.y xkbparse.y

Compiling Xfree86 4.2.0 and 4.2.1 with Bison 1.50 will give the following
errors. parse error, unexpected ":", expecting ";" or "|" and stops
processing. This patch corrects this error.

In gram.y another error exists stating type clash(`'`num') on default action
and stops processing. This patch corrects this error.

In xkbparse.y another error exist stating parse error, unexpect "|". This
patch also corrects this error.

Patch:

diff -uNr xc.orig/lib/Xft/xftgram.y xc/lib/Xft/xftgram.y
--- xc.orig/lib/Xft/xftgram.y	2001-05-16 03:32:54.000000000 -0700
+++ xc/lib/Xft/xftgram.y	2002-10-11 14:00:14.000000000 -0700
@@ -165,6 +165,7 @@
 		    matrix.yx = $4;
 		    matrix.__REALLY_YY__ = $5;
 		}
+	;
 number	:   INTEGER
 		{ $$ = (double) $1; }
 	|   DOUBLE
diff -uNr xc.orig/programs/twm/gram.y xc/programs/twm/gram.y
--- xc.orig/programs/twm/gram.y	2001-12-14 12:01:07.000000000 -0800
+++ xc/programs/twm/gram.y	2002-10-11 14:00:14.000000000 -0700
@@ -458,7 +458,7 @@
 					    ParseError = 1;
 					  }
 					}
-		  win_color_list
+		  win_color_list	{/* No action. */;}
 		| CKEYWORD string	{ if (!do_color_keyword ($1, color,
 								 $2)) {
 					    twmrc_error_prefix();
@@ -654,6 +654,8 @@
 					  RemoveDQuote(ptr);
 					  $$ = ptr;
 					}
+		;
+
 number		: NUMBER		{ $$ = $1; }
 		;
 
diff -uNr xc.orig/programs/xkbcomp/xkbparse.y xc/programs/xkbcomp/xkbparse.y
--- xc.orig/programs/xkbcomp/xkbparse.y	2001-01-17 15:45:45.000000000 -0800
+++ xc/programs/xkbcomp/xkbparse.y	2002-10-11 14:01:05.000000000 -0700
@@ -448,7 +448,6 @@
 
 ShapeDecl	:	SHAPE String OBRACE OutlineList CBRACE SEMI
 			{ $$= ShapeDeclCreate($2,(OutlineDef *)&$4->common); }
-		;
 		|	SHAPE String OBRACE CoordList CBRACE SEMI
 			{ 
 			    OutlineDef *outlines;
@@ -511,7 +510,7 @@
 			{ 
 			    $$= (OverlayKeyDef *)
 				AppendStmt(&$1->common,&$3->common);
-			};
+			}
 		|	OverlayKey
 			{ $$= $1; }
 		;
diff -uNr xc.orig/programs/xkbevd/cfgparse.y xc/programs/xkbevd/cfgparse.y
--- xc.orig/programs/xkbevd/cfgparse.y	2001-01-17 15:46:07.000000000 -0800
+++ xc/programs/xkbevd/cfgparse.y	2002-10-11 14:00:14.000000000 -0700
@@ -173,6 +173,7 @@
 		|	SHELL	 { $$ = ShellAction; }
 		|	SOUND	 { $$ = SoundAction; }
 		|		 { $$ = UnknownAction; }
+		;
 
 OptNameSpec	:	NameSpec { $$= $1; }
 		|		 { $$= NULL; }
